TVnMoviesApp/Controller/Home/HomeViewController.swift

```diff
@@ -31,17 +35,46 @@ class HomeViewController: UIViewController {
         layout.minimumLineSpacing = 0
         collectionView.register(UINib(nibName: "HomeCollectionViewCell", bundle: nil), forCellWithReuseIdentifier: "homeCell")
         activityLoading.hidesWhenStopped = true
+        
+        searchTextField.rx.text.orEmpty.asObservable()
+            .debounce(DispatchTimeInterval.milliseconds(500), scheduler: MainScheduler.instance)
+            .distinctUntilChanged()
+            .filter{ !$0.isEmpty }
+            .subscribe {
+                print($0.element ?? $0)
+                self.activityLoading.startAnimating()
+                    if var result = $0.element {
+                        result = String(result.map {
+                            $0 == " " ? "/" : $0
+                        })
+                        DispatchQueue.main.asyncAfter(deadline: .now() + 3) { [weak self] in
+                            self!.activityLoading.stopAnimating()
+                            switch self?.pickSegmentControl.selectedSegmentIndex {
+                            case 0:
+                                self?.manager.performRequestMovieSeries(idOfMovieOrTV: result, typeToWatch: "movie")
+                            case 1:
+                                self?.manager.performRequestMovieSeries(idOfMovieOrTV: result, typeToWatch: "tv")
+                            default:
+                                print(Error.self)
+                            }
+                            repeat {
```

Comment:
this is a bad thing to do, cause you'll block your main thread and the app will become unresponsive.

If you go to your Manager class you'll see, that URLSession's data task gives you a completion handler. The same technique could be used here to know when your request has finished and then you'd update your data model and call reloadData.

 Alternatively you could use more Rx-ish approach and flatMapLatest your search text observable to API request observable and subscribe to it.